llama-index-integrations/vector_stores/llama-index-vector-stores-databricks/llama_index/vector_stores/databricks/base.py

```diff
@@ -333,17 +346,30 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
         for result in search_resp.get("result", {}).get("data_array", []):
             doc_id = result[columns.index(self._primary_key)]
             text_content = result[columns.index(self.text_column)]
-            metadata = {
+
+            # extract metadata and node_info respectively from metadata_dict
+            metadata_dict = {
                 col: value
                 for col, value in zip(columns[:-1], result[:-1])
-                if col not in [self._primary_key, self.text_column]
+                if col not in [self._primary_key, self.text_column, "doc_id"]
             }
-            metadata[self._primary_key] = doc_id
+            metadata = {
+                col: value
+                for col, value in metadata_dict.items()
+                if col not in ["node_info"]
```

Comment:
```suggestion
                if col != "node_info"
```

Reply:
Changes made as suggested in "minor fixes"